.github/workflows/pr.yml

```
@@ -39,12 +41,36 @@ jobs:
         run: |
           echo "systems=$(jq -c <ci/supportedSystems.json)" >> "$GITHUB_OUTPUT"
 
+      - name: Determine branch type
+        id: types
+        uses: actions/github-script@60a0d83039c74a4aee543508d2ffcb1c3799cdea # v7.0.1
+        with:
+          script: |
+            const base = Object.entries({
+              channel: /^(nixos-|nixpkgs-)/.test(context.payload.pull_request.base.ref),
+              stable: /^(release-|staging-(next-)?)\d+.\d+$/.test(context.payload.pull_request.base.ref),
+              staging: !/^master$|^release-/.test(context.payload.pull_request.base.ref)
```

Won't _"not `master` or `release-*`"_ false positive for a bunch of branches that _aren't_ related to the staging cycle?

---

The sheer number of branch names (by the number of types) is making me wonder if we need a unit test for these regexes. That would specify which "types" we expect for a bunch of example branch names.

That'd require extracting the logic out of the workflow though... Or _at least_ extracting the patterns into a separate file, to test them in isolation.

At first glance such a hard coded "expect" mapping might sound like it makes the regex patterns redundant, however I think they are still useful for when new branches (and new versions) are added.

---

> Well, it's not really important whether they are related to the staging _cycle_. The important bit is, that we don't build the NixOS manual on anything that is not master/release. Only on those branches, we are confident that building the nixos manual will not cause a huge number of rebuilds. _Any_ other branch will suffer from potentially having a huge rebuild in them, which has not been built by hydra, yet.

Gotcha. That makes sense.

> But I gave it a name. And I called it staging... because I just couldn't come up with a name that reflects master+release branches well. Any idea?

Nothing harder than choosing a name!

We could try to indicate that this is something _other_ than the set of "known" branches we want to build on:

- other
- unknown
- undefined
- misc

---

Just to double check on the "false positives" angle; _channel_ branches will currently also match this category of "not `master` or `release-*`". Is that intended/unimportant, or should it be:

```suggestion
              staging: !/^master$|^release-|^nixos-|^nixpkgs-/.test(context.payload.pull_request.base.ref)
```

?

---

I guess I was focusing on ensuring the type names are well suited to their matched branches, because it would be confusing if we reuse these branch-types elsewhere in the workflows, perhaps for a different reason.

I agree it doesn't matter too much for the current usage, but could lead to confusion down the line.

Same argument for the channels type being mutually exclusive with the other types. Not important now, but could be confusing later.

Would "staging-like" be a fair compromise? That way the implementation can stay for now and the intent can be reflected in the name, along with implying that it also matches others.

Your call.

---

SGTM 👍